cypress-io/cypress
# chore: initial protocol api work

### Additional details


This is an initial base implementation of hooking into the protocol API that will be delivered from the cloud. It hooks into the lifecycle of specs, tests, and browser connections.

### Steps to test


- Clone the `cypress-services` repo
  - Run `yarn`
  - Run `yarn watch` in `packages/app-capture-protocol`
- Set `CYPRESS_LOCAL_PROTOCOL_PATH` to the path to `cypress-services/packages/app-capture-protocol/dist/index.js`
- Clone the `cypress` repo
  - Run `yarn`
  - Execute `cypress run` on a project in record mode with `DEBUG` set to `cypress:*protocol*` and ensure you can see the lifecycle events

### How has the user experience changed?


n/a

### PR Tasks


- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
